src/main/java/me/itzg/helpers/modrinth/ModrinthApiPackFetcher.java

```diff
@@ -0,0 +1,82 @@
+package me.itzg.helpers.modrinth;
+
+import java.nio.file.Path;
+
+import lombok.extern.slf4j.Slf4j;
+import me.itzg.helpers.errors.*;
+import me.itzg.helpers.files.Manifests;
+import me.itzg.helpers.http.FailedRequestException;
+import me.itzg.helpers.modrinth.model.*;
+import reactor.core.publisher.Mono;
+
+@Slf4j
+public class ModrinthApiPackFetcher implements ModrinthPackFetcher {
+    private ModrinthApiClient apiClient;
+
+    private ProjectRef modpackProjectRef;
+    private Mono<Version> modpackVersionStream;
+
+    private Loader modLoaderType;
+    private String gameVersion;
+    private VersionType defaultVersionType;
+    private Path modpackOutputDirectory;
+
+    ModrinthApiPackFetcher(InstallModrinthModpackCommand config) {
+        this.apiClient = new ModrinthApiClient(
+            config.baseUrl, "install-modrinth-modpack",
+            config.sharedFetchArgs.options());
```

Comment:
Don't create a new instance of this. There should be one for the whole command-call -- the pass the existing one in much like dependency injection like Spring, etc.